### DISCUSSION TITLE
Grep with output variables

### DISCUSSION THREAD

**Discussion by jonny7** [community/community#38216] [Actions] [answered]
I'm trying to grep for the latest tag like so

```
      - name: Get release tag
        id: version
        run: |
          BRANCH=$(git rev-parse --abbrev-ref HEAD)
          echo $BRANCH
          echo "::set-output name=tag::$(echo $BRANCH | sed 's/release\///')"

      - name: build latest tag
        id: latest
        run: |
          REGEX=^${{ steps.version.outputs.tag }}-dev.+$
          echo $REGEX
          LATEST=$(git tag | grep -E $REGEX | tail -n1)
          echo $LATEST
          echo "::set-output name=tag_name::$(echo $LATEST)"
```

But, this seems to always return empty, even though when I run it with `nektos/act` it returns the last matching tag. I've tried a few different interpolation ways, but I'm not sure what is wrong and why act works but actions doesn't seem to

**Comment by david-engelmann** [comment]
Is it installed? What does this run on?

**Reply by jonny7** [reply]
It runs on ubuntu-latest, i'll check the installation for that

**Reply by jonny7** [reply]
It should be available according to this: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

**Reply by david-engelmann** [reply]
It's present in Ubuntu

**Comment by airtower-luna** [comment, marked as answer]
It depends on the context of your workflow, but most likely the Git repo on the runner doesn't have that information. By default `actions/checkout` gets only the current commit.

You can tell `actions/checkout` to fetch the whole history or add a suitable `git fetch` command, but depending on what exactly you're trying to do it might be much easier to get the tag from context (e.g. if your workflow runs only on tag push, use `github.ref_name`).

**Reply by jonny7** [reply]
Thanks, I just figured this out, it didn't have the tags so I added `git fetch --prune --unshallow --tags` to bring those in - thanks both 🙂
